ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -434,6 +448,187 @@ traceLedgerProcessing
       mapM_ Debug.traceMarkerIO traces
       return $ newLedger
 
+{-------------------------------------------------------------------------------
+  Analysis: maintain a ledger state and time the five major ledger calculations
+  for each block:
+
+  0. Forecast.
+  1. Header tick.
+  2. Header application.
+  3. Block tick.
+  4. Block application.
+
+  We focus on these 5 operations because they are involved in:
+
+  - Chain syncing.
+  - Block forging.
+  - Block validation.
+
+-------------------------------------------------------------------------------}
+
+data BenchmarkLedgerOpsState blk = BenchmarkLedgerOpsState {
+       -- | RTS stats that correspond to either the point in time in which the
+       -- benchmarking process started, or the end of the previous block
+       -- application.
+       prevRtsStats    :: !GC.RTSStats
```

Comment:
I think we could drop this state data type if we call `GC.getRTSStats` six times, once as the very first and once as the very last thing (other than subtractions and rendering) in each application of `process`. And that would avoid measuring the costs of everything (eg `processAll` etc) other than the 5 steps done within `process`.